docs/CONNECTOR_PROTOCOL.md

```diff
@@ -82,29 +110,111 @@ This is our main communication index, used to communicate the connector's config
+    },
+    "version" : "1"
+  },
   "properties" : {
     "api_key_id" : { "type" : "keyword" },
     "configuration" : { "type" : "object" },
     "description" : { "type" : "text" },
-    "error" : { "type" : "text" },
+    "error" : { "type" : "keyword" },
+    "filtering" : {
+      "properties" : {
+        "domain" : { "type" : "keyword" },
+        "active" : {
+          "properties" : {
+            "rules" : {
+              "properties" : {
+                "id" : { "type" : "keyword" },
+                "policy" : { "type" : "keyword" },
+                "field" : { "type" : "keyword" },
+                "rule" : { "type" : "keyword" },
+                "value" : { "type" : "keyword" },
+                "order" : { "type" : "short" },
+                "created_at" : { "type" : "date" },
+                "updated_at" : { "type" : "date" }
+              }
+            },
+            "advanced_snippet" : {
+              "properties" : {
+                "value" : { "type" : "object" },
+                "created_at" : { "type" : "date" },
+                "updated_at" : { "type" : "date" }
+              }
+            },
+            "validation" : {
+              "properties" : {
+                "state" : { "type" : "keyword" },
+                "errors" : {
+                  "properties" : {
+                    "ids" : { "type" : "keyword" },
+                    "messages" : { "type" : "text" }
+                  }
+                }
+              }
+            }
+          }
+        },
+        "draft" : {
+          "properties" : {
+            "rules" : {
+              "properties" : {
+                "id" : { "type" : "keyword" },
+                "policy" : { "type" : "keyword" },
+                "field" : { "type" : "keyword" },
+                "rule" : { "type" : "keyword" },
+                "value" : { "type" : "keyword" },
+                "order" : { "type" : "short" },
+                "created_at" : { "type" : "date" },
+                "updated_at" : { "type" : "date" }
+              }
+            },
+            "advanced_snippet" : {
+              "properties" : {
+                "value" : { "type" : "object" },
+                "created_at" : { "type" : "date" },
+                "updated_at" : { "type" : "date" }
+              }
+            },
+            "validation" : {
+              "properties" : {
+                "state" : { "type" : "keyword" },
+                "errors" : {
+                  "properties" : {
+                    "ids" : { "type" : "keyword" },
+                    "messages" : { "type" : "text" }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
     "index_name" : { "type" : "keyword" },
     "is_native" : { "type" : "boolean" },
     "language" : { "type" : "keyword" },
     "last_seen" : { "type" : "date" },
-    "last_sync_error" : { "type" : "text" },
+    "last_sync_error" : { "type" : "keyword" },
```

Comment:
The mapping was previously defined as Keyword, and this document was just out of sync with ent-search and kibana.